### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by ayooshkathuria**
The purpose of this branch is to add python 3 support for this implementation. There are several compatibility issues which prevent this implementation from successfully executing on python 3. 

1. A lot of code doesn't work because of the way Python 2 and Python 3 handle the `/` (division) operator. The operator performs integer division in Py2, whereas it performs float division in Py3. This causes a lot of problems, because a lot of parameters (like padding, number of epochs, stride, anchor step) need to be integers, but the division operator in Python 3 yields floats. For all these instances, I have found, I've replaced the `/` operator with `//` operator, which performs integer division in both Py2 and Py3. There was an instance where such a thing didn't work (I printed the python type of the parameter before it was passed into the function which was raising the error, it was float), so I had to explicitly typecast the parameter as an int.

2. Some functions that are used exist in python 2, but do not exist in python 3 anymore. In this case, the functions are `xrange` and `cPickle`. I've introduced these issues by using try catch blocks. 

3. There is something about reading and writing files using python's `open` module that works fine with the current implementation in Python2, but the same function throws errors when the code is run in Python 3. These include defining the file to be read as a byte file, and then trying to read a file of the disk that the interpreter recognizes as a string (char?) file. I've simply followed the errors, and fixed the errors by matching up the type of file being read/written and the argument passed to `open`

4. Lack of parenthesis with the `print` function. 

Now, I have not tested the code in it's entirety, as in I've tested only the following parts of code. 

1. Running a test detection `detect.py`
2. Training with `train.py` with ONLY Pascal VOC 2007/2012 database. 
3. Evaluating using `valid.py` and `scripts/voc_eval.py`. I got an mAP of 75.13 with the pre-trained `yolo-voc.weights`

I'm wary about the fact that if compatibility issues are present in code **ONLY** used during COCO training, I will have missed it. I intend to test the code using COCO, but that won't be possible for another week. 

I've tested the code on Python 3.6.3, on a nVidia Tesla K80, and nVidia Quadro K1200 (I coudn't get the network to train on K1200 because the batch size was too big to fit into VRAM, and I was being lazy).

### CODE REVIEW

**Comment by ayooshkathuria** [comment]
The same issues have been addressed in #58 .
